frontend/src/store/features/gov/govSlice.ts

```diff
@@ -0,0 +1,676 @@
+      errMsg: '',
+      proposals: [],
+      pagination: {
+        next_key: '',
+        total: '',
+      },
+    },
+    depositParams: {
+      status: TxStatus.INIT,
+      errMsg: '',
+      params: {
+        min_deposit: [
+          {
+            denom: '',
+            amount: '',
+          },
+        ],
+        max_deposit_period: '',
+      },
+    },
+    votes: {
+      status: TxStatus.INIT,
+      errMsg: '',
+      proposals: {},
+    },
+    tally: {
+      status: TxStatus.INIT,
+      errMsg: '',
+      proposalTally: {},
+    },
+    tx: {
+      status: TxStatus.INIT,
+      txHash: '',
+    },
+  },
+  proposalInfo: {
+    status: TxStatus.INIT,
+    errMsg: '',
+  },
+};
+
+export const getProposal = createAsyncThunk(
+  'gov/proposal-info',
+  async (data: GetProposalInputs, { rejectWithValue }) => {
+    try {
+      const response = await govService.proposal(data.baseURL, data.proposalId);
+      return {
+        chainID: data.chainID,
+        data: response.data,
+      };
+    } catch (error) {
+      if (error instanceof AxiosError)
+        return rejectWithValue({ message: error.message });
+      return rejectWithValue({ message: ERR_UNKNOWN });
+    }
+  }
+);
+
+export const getProposalsInDeposit = createAsyncThunk(
+  'gov/deposit-proposals',
+  async (data: GetProposalsInDepositInputs, { rejectWithValue, dispatch }) => {
+    try {
+      const response = await govService.proposals(
+        data.baseURL,
+        data?.key,
+        data?.limit,
+        PROPSAL_STATUS_DEPOSIT
+      );
+
+      if (response?.data?.proposals?.length && data?.chainID?.length) {
+        dispatch(
+          getDepositParams({
+            baseURL: data.baseURL,
+            chainID: data.chainID,
+          })
+        );
+      }
+
+      return {
+        chainID: data.chainID,
+        data: response.data,
+      };
+    } catch (error) {
+      if (error instanceof AxiosError)
+        return rejectWithValue({ message: error.message });
+      return rejectWithValue({ message: ERR_UNKNOWN });
+    }
+  }
+);
+
+export const getDepositParams = createAsyncThunk(
+  'gov/deposit-params',
+  async (data: GetDepositParamsInputs, { rejectWithValue }) => {
+    try {
+      const response = await govService.depositParams(data.baseURL);
+
+      return {
+        chainID: data.chainID,
+        data: response.data,
+      };
+    } catch (error) {
+      if (error instanceof AxiosError)
+        return rejectWithValue({ message: error.message });
+      return rejectWithValue({ message: ERR_UNKNOWN });
+    }
+  }
+);
+
+export const getProposalsInVoting = createAsyncThunk(
+  'gov/active-proposals',
+  async (data: GetProposalsInVotingInputs, { rejectWithValue, dispatch }) => {
+    try {
+      const response = await govService.proposals(
+        data.baseURL,
+        data.key,
+        data.limit,
+        PROPOSAL_STATUS_ACTIVE
+      );
+
+      if (response?.data?.proposals?.length) {
```

Comment:
its better to write to like this 
```suggestion
     const { data } = response || {};
const proposals = data?.proposals || [];

proposals.forEach((proposal) => {
  const proposalId = Number(proposal.proposal_id);

  dispatch(getProposalTally({
    baseURL: data?.baseURL,
    proposalId,
    chainID: data?.chainID,
  }));

  dispatch(getVotes({
    baseURL: data?.baseURL,
    proposalId,
    voter: data?.voter,
    chainID: data?.chainID,
  }));
});

```